apps/science/src/main.cpp

```diff
@@ -1,19 +1,147 @@
+Motor indexerMotor(MTR_PWM_1, MTR_DIR_1, false);
+
+// Servo
+LimServo coverServo(SRVO_PWM_1, ScienceConfig::coverServoRange, ScienceConfig::coverServoMaxPulse,
+                    ScienceConfig::coverServoMinPulse);
+
+LimServo diggerServo(SRVO_PWM_2, ScienceConfig::diggerServoRange, ScienceConfig::diggerServoMaxPulse,
+                     ScienceConfig::diggerServoMinPulse);
+
+// encoders
+EncoderAbsolute_PWM elevatorEncoder(ScienceConfig::elevatorEncoderConfig);
+EncoderRelative_Quadrature indexerEncoder(ScienceConfig::centrifugeEncoderConfig);
+
+// limit switches
+DigitalIn elevatorLimTop(LIM_SW_3);
+DigitalIn elevatorLimBottom(LIM_SW_4);
+DigitalIn indexerLimFront(LIM_SW_1);
+DigitalIn indexerLimBack(LIM_SW_2);
+
+// Actuators
+ActuatorController elevatorActuator(ScienceConfig::diggerLiftActuatorConfig, elevatorMotor, elevatorEncoder,
+                                    elevatorLimBottom, elevatorLimTop);
+ActuatorController indexerActuator(ScienceConfig::indexerActuatorConfig, indexerMotor, indexerEncoder, indexerLimFront,
+                                   indexerLimBack);
+
+// I2C
+MoistureSensor moistureSensor = MoistureSensor(TEMP_MOIST_I2C_SDA, TEMP_MOIST_I2C_SCL);
+
+// LEDs
 DigitalOut led1(LED1);
+DigitalOut ledR(LED_R);
+DigitalOut ledG(LED_G);
+DigitalOut ledB(LED_B);
 
-constexpr auto k_interval = 500ms;
+// Declaring can object
+CAN can(CAN1_RX, CAN1_TX, HWBRIDGE::ROVERCONFIG::ROVER_CANBUS_FREQUENCY);
 
-CANBus can(CAN1_RX, CAN1_TX, HWBRIDGE::ROVERCONFIG::ROVER_CANBUS_FREQUENCY);
+// static objects
+static mbed_error_status_t setMotionData(CANMsg &msg) {
+  float motionData;
+  msg.getPayload(motionData);
+
+  switch (msg.getID()) {
+    case HWBRIDGE::CANID::SET_GENEVA_INDEX:
+      return indexerActuator.setMotionData(motionData);
+    case HWBRIDGE::CANID::SET_SCOOPER_ANGLE:  // not sure what this case does will contact younes soon to discuss
+      return elevatorActuator.setMotionData(motionData);  // with him (perhaps this is elevatorActuator??)
+    case HWBRIDGE::CANID::SET_COVER_ANGLE:
+      return coverServo.setPosition(motionData);
+    case HWBRIDGE::CANID::SET_ELEVATOR_HEIGHT:
+      return diggerServo.setPosition(motionData);
+    default:
+      return MBED_ERROR_INVALID_ARGUMENT;
+  }
+}
+
+static CANMsg::CANMsgHandlerMap canHandleMap = {{HWBRIDGE::CANID::SET_GENEVA_INDEX, setMotionData},
+                                                {HWBRIDGE::CANID::SET_SCOOPER_ANGLE, setMotionData},
+                                                {HWBRIDGE::CANID::SET_COVER_ANGLE, setMotionData},
+                                                {HWBRIDGE::CANID::SET_ELEVATOR_HEIGHT, setMotionData}};
+
+// CAN Threads
+Thread rxCANProcessorThread;
+Thread txCANProcessorThread;
+
+// recieving CAN messages
+void rxCANProcessor() {
+  CANMsg rxMsg;
+  while (true) {
+    if (can.read(rxMsg)) {
+      if (canHandleMap.size() > 0) {
+        canHandleMap[rxMsg.getID()](rxMsg);
+      } else {
```

Comment:
this is the old way we used to process incoming can msgs. heres the new way. also remove the else that does nothign
if (can1.read(rxMsg)) {
      canHandlerMap.at(rxMsg.getID())(rxMsg);
    }

Reply:
done